### PR TITLE
LDAP: Add fallback to search_base_dns if group_search_base_dns is undefined.

### LINKED ISSUE #20862: Point out that 'group_search_base_dns' is necessary for group search feature!
**What happened**:
I tried to configure nested ldap groups in AD according to this:
https://github.com/grafana/grafana/blob/master/docs/sources/auth/ldap.md#nestedrecursive-group-membership

**What you expected to happen**:
I expected the nested groups to return as configured in our AD.

**How to reproduce it (as minimally and precisely as possible)**:
Configure as descripted in the docs with a Microsift AD.

**How to fix it**
The problem was that `group_search_base_dns` was **NOT** set and then the query is **not executed**

Two suggestions:
  * Update the documentation accordingly and point out that `group_search_base_dns` is necessary! https://github.com/grafana/grafana/blob/b20a258b72d3130747118c6f3c535fc5d1b3a15f/pkg/services/ldap/ldap.go#L481
  * add a fallback that if `GroupSearchBaseDNs` is *nil* but `GroupSearchFilter` is set then use `SearchBaseDNs`

Also this Config seems to work out of the "box" (without adding own OUs)
```
group_search_filter = "(member:1.2.840.113556.1.4.1941:=%s)"
group_search_base_dns = ["DC=mycorp,DC=mytld"]
group_search_filter_user_attribute = "dn"
```
**Anything else we need to know?**:
I love Grafana!

**Environment**:
- Grafana version: 6.5.1
- Data source type & version: InfluxDB
- OS Grafana is installed on: CentOS 7.7
- User OS & Browser: Chrome
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by bb-Ricardo**
**Which issue(s) this PR fixes**:

This PR adds a fallback to `search_base_dns` for `group_search_base_dns` in ldap.toml. This will add better support for nested groups configurations.

Fixes #20862

**Special notes for your reviewer**:
Theses are my first steps in Go. Please review properly.

### CODE REVIEW

File: docs/sources/auth/ldap.md [hunk 1]
```diff
@@ -199,6 +201,13 @@ group_search_filter = "(member:1.2.840.113556.1.4.1941:=CN=%s,[user container/OU
 group_search_filter = "(|(member:1.2.840.113556.1.4.1941:=CN=%s,[user container/OU])(member:1.2.840.113556.1.4.1941:=CN=%s,[another user container/OU]))"
 group_search_filter_user_attribute = "cn"
 ```
+
+This config should work out of the box
```
Comment (oddlittlebird) [inline docs/sources/auth/ldap.md]: What should this configuration do? I suggest something like:

To define a new group search filter:

Avoid idioms like "out of the box" which might be difficult for non-native English speakers to understand.

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

File: docs/sources/auth/ldap.md [hunk 2]
```diff
@@ -189,6 +189,8 @@ Setting | Required | Description | Default
 Users with nested/recursive group membership must have an LDAP server that supports `LDAP_MATCHING_RULE_IN_CHAIN`
 and configure `group_search_filter` in a way that it returns the groups the submitted username is a member of.
 
+To configure `group_search_filter` you can use `group_search_base_dns` to define where the matching groups are defined. Otherwise the previously defined `search_base_dns` will be used.
```
[the remark this answers is not in the document]
Comment (bb-Ricardo) [reply docs/sources/auth/ldap.md]: changed in lates commit

File: docs/sources/auth/ldap.md (hunk 1 above)
Comment (bb-Ricardo) [reply docs/sources/auth/ldap.md]: changed in lates commit

**Review by oddlittlebird — APPROVED** [review]

**Comment by oddlittlebird** [comment]
Thanks for contributing to Grafana!

**Pull request merged**